Handling failed load

dagzn opened the pull request:
## Error handling for loading an event list
Handling the cases where an elements data fails to load. We display the error in the detail-list and also do not show its groups or items in the timeline.

Fixed the issue where we load the data once again each time we switch the tabs Correlated and All. Now the same data is used once again and we only update it when the date range changes.

jeffj6123 commented on src/SfxWeb/src/app/modules/detail-list-templates/detail-list/detail-list.component.html:
```diff
@@ -88,12 +88,15 @@
                         <td [attr.colspan]="listSettings.columnSettings.length">No items to display.</td>
                     </tr>
+                    <tr *ngIf="!successfulLoad">
+                        <td [attr.colspan]="listSettings.columnSettings.length">Error: The list of events has failed to load.</td>
```
change this to a more generic phrase, like

"Items failed to load"

jeffj6123 commented:
So a good thing to do would be show something in the tabs to make it more obvious instead of needing to go to the tab's list

dagzn commented:
> So a good thing to do would be show something in the tabs to make it more obvious instead of needing to go to the tab's list

Just added some more info regarding that part. Added some icons to show which tabs have an error and displayed a warning stating that some items failed to load. My only concern is that users might think that the error icon is a closing button, let me know what you think about this issue @jeffj6123 

![image](https://user-images.githubusercontent.com/26049483/122591065-0836a400-d028-11eb-946e-17f404fa72e1.png)

jeffj6123 commented on src/SfxWeb/src/app/modules/event-store/event-store/event-store.component.html:
```diff
@@ -44,18 +44,23 @@
         </div>
         <br>
 
+        <app-health-badge *ngIf="failedRefresh" [text]="' Some items failed to load.'" [showText]="true" badgeClass="badge-warning"></app-health-badge>
```
remove the []s around text and showText

jeffj6123 commented on src/SfxWeb/src/app/modules/event-store/event-store/event-store.component.html:
```diff
@@ -44,18 +44,23 @@
             <div ngbNav #nav="ngbNav">
                 <div ngbNavItem *ngFor="let data of listEventStoreData">
                     <a ngbNavLink class="bar-name">
+                        <app-health-badge *ngIf="!data.eventsList.lastRefreshWasSuccessful" 
```
remove the []s around text and showText
